functions/visualizations.py

```diff
@@ -105,7 +105,17 @@ def _add_antibody_metadata(
         ct_md = TrialMetadata.find_by_trial_id(
             file_record.trial_id, session=session
         ).metadata_json
-    assay_md = ct_md.get("assays", {}).get(upload_type, {})
+
+    assay_instances = ct_md.get("assays", {}).get(upload_type, [])
+    # asserting that this will return a list, which is not necessarily true
+    # check cidc-schemas/schemas/assays/components/available_assays.json
+
+    if isinstance(assay_instances, dict):
+        # only exception to list, eg olink
+        assay_md = assay_instances
+    else:
+        # ASSUME that this is the most recent assay
+        assay_md = assay_instances[-1] if len(assay_instances) else {}
```

Comment:
Is there reason to believe this is a fair assumption? What are the implications of this assumption being wrong?

Reply:
so I looked into looking through the different instances to find the file that this is being called on, but I couldn't think of a simple to implement way

Reply:
So, I think we actually need to find a way to do just that. Finding the most recent assay upload metadata is isn't enough, since it's possible that the file we're operating on is an old file from a previous upload (e.g., if we're re-running processing on it). Happy to discuss possible solutions for this offline.

Reply:
add in following commit